lib/JSON/Validator.pm

```diff
@@ -586,13 +586,15 @@ sub _validate_all_of {
 
   return if not @errors;
 
+  my @details = map { map $_->details, @{$_}[1 .. $#{$_}] } @errors;
+
   return prefix_errors(allOf => @errors)
-    if @errors == 1
-    or grep $_->[1]->details->[1] ne 'type', @errors;
+    if @details == 1
+    or grep $_->[1] ne 'type', @details;
 
   # combine all 'type' errors together
-  my $want_types = join('/', uniq map $_->[1]->details->[0], @errors);
-  return E $path, [allOf => type => $want_types, $errors[-1][1]->details->[2]];
+  my $want_types = join('/', uniq map $_->[0], @details);
+  return E $path, [allOf => type => $want_types, $details[-1]->[2]];
```

Comment:
No need for the last `->`, and `join` does not need any parenthesis. (Same below)

Reply:
fixed.